### ISSUE TITLE
Sources can be duplicated when similar sources for different chunks on !sources command

### ISSUE THREAD

**Issue by bolinocroustibat** [etalab-ia/albert-tchapbot#39]
We should de-duplicate similar sources.

<img width="862" alt="Screenshot 2024-05-26 at 18 50 27" src="https://github.com/etalab-ia/albert-tchapbot/assets/4353767/6715742b-9453-4853-8352-39fa3a862e1e">

**Comment by dtrckd** [comment]
Hey, I don't see any duplicated sources. 
They are duplicated sheets (e.g `F[0-9]{6}`), but sources are derived from chunks and sheets can contains multiple chunks. Si it is not surprising to find several times the same sheets referenced. But if you look at the context (the text in parenthesis, they are all different. The context is like a breadcrumb of the chunk inside the sheet. And finally, there is no direct links to chunks as they come from the same sheets, which explain why there are the se same links.

**Comment by dtrckd** [comment]
Ah, I missed the one that are actually real duplicated.

**Comment by bolinocroustibat** [comment]
@pedevineau Can you confirm you're OK with the current !source behaviour?
@dtrckd This was open after some user's feedback, might be better to reopen while we make sure we all agree on the current behaviour

**Comment by dtrckd** [comment]
What we can do is to add an anchor in link for different chunks. For example 

```
https://www.service-public.fr/particuliers/vosdroits/F59#chunk_1
```

Even if the anchor does not exist, it can give a hint of the user of why this is the actual same URL.

Let me know if you have better idea.

**Comment by pedevineau** [comment]
How do we choose the titles related to chunks? My suggestion would be: let us return the title of the sheet once with the url. So it will be easy to deduplicate

**Comment by pedevineau** [comment]
Yes I know there is no deduplicates of chunks, I was considering dedupling sheets, because at the end every url targets the same page. The anchor system doesn't work in general, because our chunks are not always related to the DILA webpage anchors, are them?

**Comment by dtrckd** [comment]
Yes, you're right, the anchor idea was just to give a visual hint.
